### PR TITLE
[qol 1/n]: utils

### PR DESCRIPTION
**Pull request by jonathanlastmileai**
[qol 1/n]: utils

Summary:

Adding some very fundamental utilities to make development smoother, e.g.
* safe file I/O
* abstracted config class construction from CLi
* Result convenience functions
etc.

### CODE REVIEW

File: CODEOWNERS
Comment (rholinshead) [inline CODEOWNERS:1]:
```suggestion
* @saqadri @rholinshead @suyoglastmileai @Ankush-lastmile @jonathanlastmileai @rossdanlm
```

Looks like Ankush updated his username and accidental `rite` suffix on Rossdan's

**Review by rholinshead — APPROVED** [review]
Accepting to unblock but please fix the CODEOWNERS file before landing

**Comment by jonathanlastmileai** [comment]
> Accepting to unblock but please fix the CODEOWNERS file before landing

Thanks, good catch

**Pull request merged**
